### DISCUSSION TITLE
How to quit test execution without hampering report generation in Taiko gauge

### DISCUSSION THREAD

**Discussion by Ananda0488** [getgauge/taiko#2347] [General]
I have a requirement as follows:
My test script consist of 5 steps (1,2,3,4,5). In step 2 I am checking a condition. 
If the condition is TRUE, the remaining steps (3,4,5) will be skipped, the test will end, should pass and report will be generated. My question: Is there any way to do this?
I used 'process.exit(1)' in step 2 as follows, but that is not skipping (3,4,5).

step("Step 2", async function() {	
	if(condition==true){
             process.exit(1)
        }
});

**Comment by zabil** [comment]
Gauge looks for an error thrown out from step 2 to skip the rest of the steps. At the moment I do not think there's a way to skip or stop execution unless there's an error.

**Reply by Ananda0488** [reply]
ok.. thank you Zabil :) 
I didn't notice i put category of this discussion as Development, just changing it to General.
